packages/plugin-ext-vscode/src/node/plugin-vscode-decompress.ts

```diff
@@ -0,0 +1,36 @@
+// *****************************************************************************
+// Copyright (C) 2023 EclipseSource and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import * as decompress from 'decompress';
+import * as path from 'path';
+import * as fs from '@theia/core/shared/fs-extra';
+
+export async function decompressExtension(sourcePath: string, destPath: string): Promise<boolean> {
+    try {
+        await decompress(sourcePath, destPath);
+        if (sourcePath.endsWith('.tgz')) {
+            const extensionPath = path.join(destPath, 'package');
+            const vscodeNodeModulesPath = path.join(extensionPath, 'vscode_node_modules.zip');
+            if (await fs.pathExists(vscodeNodeModulesPath)) {
+                await decompress(vscodeNodeModulesPath, path.join(extensionPath, 'node_modules'));
+            }
+        }
+        return true;
+    } catch (error) {
+        console.error(`Failed to decompress ${sourcePath} to ${destPath}: ${error}`);
```

Comment:
Why are we not just letting the exception propagate?

Reply:
Sure, stupid mistake.